src/Server.cpp

```diff
@@ -1,181 +1,294 @@
+ * It checks for duplicate servers using checkDuplicateServer() and
+ * skips opening the socket server if it already exists.
+ *
+ * @return true if at least one virtual server was added, false otherwise.
  */
-Server& Server::operator=(const Server& ref)
+bool Server::initVirtualServers()
 {
-	if (this == &ref)
-		return *this;
+	LOG_INFO << "Initializing virtual servers";
 
-	m_serverSock = ref.m_serverSock;
-	m_epfd = ref.m_epfd;
+	for (std::vector<ServerConfig>::const_iterator iter = m_configFile.serverConfigs.begin();
+		 iter != m_configFile.serverConfigs.end(); ++iter) {
 
-	return *this;
-}
+		LOG_DEBUG << "Adding virtual server: " << iter->serverName << " on " << iter->host << ":" << iter->port;
 
-/* ====== MEMBER FUNCTIONS ====== */
+		if (checkDuplicateServer(m_virtualServers, iter->host, webutils::toString(iter->port)))
+			continue;
+
+		if (!addVirtualServer(iter->host, m_backlog, webutils::toString(iter->port))) {
+			LOG_DEBUG << "Failed to add virtual server: " << iter->serverName;
+			continue;
+		}
+	}
+
+	if (m_virtualServers.empty()) {
+		LOG_ERROR << "No virtual servers added";
+		return false;
+	}
+
+	LOG_INFO << "Finished setting up virtual servers";
+	return true;
+}
 
 /**
- * @brief Run the server event loop.
+ * @brief Add a virtual server.
  *
- * This method enters a continuous loop to handle incoming events using epoll.
- * It waits for events using epoll_wait, processes the events, and dispatches
- * them to appropriate handler functions.
+ * Creates a listening socket for a virtual server using the provided host, port, and backlog.
+ * It uses getaddrinfo() to get the address information for the provided host and port.
+ * It then creates a listening socket using createListeningSocket() and registers the virtual server
+ * using Server::registerVirtualServer().
  *
- * @throws std::runtime_error if epoll_wait encounters an error.
+ * @param host host address for the virtual server.
+ * @param backlog maximum length to which the queue of pending connections for the virtual server may grow.
+ * @param port port number for the virtual server.
  *
+ * @return true if the virtual server was successfully added, false otherwise.
  */
-void Server::run()
+bool Server::addVirtualServer(const std::string& host, const int backlog, const std::string& port)
 {
-	RequestParser parser;
+	const char* node = NULL;
 
-	while (true) {
-		struct epoll_event events[MAX_EVENTS];
-		// Blocking call to epoll_wait
-		int nfds = epoll_wait(m_epfd, events, MAX_EVENTS, -1);
-		if (nfds < 0)
-			throw std::runtime_error("epoll_wait");
-
-		for (int i = 0; i < nfds; ++i) {
-			if (events[i].data.fd == m_serverSock)
-				acceptConnection();
-			else
-				handleConnections(events[i].data.fd, parser);
-		}
+	if (!host.empty() || host != "*")
+		node = host.c_str();
+
+	struct addrinfo hints = {
+		AI_PASSIVE, /* .ai_flags - If node is NULL returned address will be suitable to bind(2) a socket which can
```

Comment:
That would just copy more stuff out of the man page of getaddrinfo().
I would see the comments as little reminders what it does / for what its for. If you want more info youd'd check the man page.

I can add a @see man getaddrinfo() for more information of the hints struct.

Reply:
I would want to avoid having to consult external resources and interrupting the flow of reading when I am trying to understand the code.

Suggestion:
> With this flag set: If node is NULL, returned address (INADDR_ANY) will be suitable to accept any network connections. W/o the flag the returned address is a loopback not allowing external connections.

Reply:
The generated doxygen didn't look nice.
I pulled out the whole getaddrinfo() part into a new function resolveListeningAddresses() and moved the comments re values of hints struct into the doxygen comment. Added your suggested part to AI_PASSIVE part.
I left little comments in the function stating only the name of the struct value (since dedicated init not possible in C++98). These don't get pulled out by doxygen